rls/src/main/java/io/grpc/rls/RlsProtoData.java

```diff
@@ -548,4 +528,19 @@ public String toString() {
       }
     }
   }
+
+  @AutoValue
+  abstract static class ExtraKeys {
+    static final ExtraKeys DEFAULT = create(null, null, null);
+
+    @Nullable abstract String host();
+
+    @Nullable abstract String service();
+
+    @Nullable abstract String method();
+
+    static ExtraKeys create(String host, String service, String method) {
```

Comment:
Should arguments also be annotated `@Nullable`?

Reply:
Done.